zone/bot_database.cpp

```diff
@@ -918,73 +918,113 @@ bool BotDatabase::LoadTimers(Bot* bot_inst)
 		return false;
 
 	query = StringFormat(
-		"SELECT"
-		" IfNull(bt.`timer_id`, '0') As timer_id,"
-		" IfNull(bt.`timer_value`, '0') As timer_value,"
-		" IfNull(MAX(sn.`recast_time`), '0') AS MaxTimer"
-		" FROM `bot_timers` bt, `spells_new` sn"
-		" WHERE bt.`bot_id` = '%u' AND sn.`EndurTimerIndex` = ("
-		"SELECT case"
-		" WHEN timer_id > '%i' THEN timer_id - '%i'"
-		" ELSE timer_id END AS timer_id"
-		" FROM `bot_timers` WHERE `timer_id` = bt.`timer_id` AND `bot_id` = bt.`bot_id`" // double-check validity
-		")"
-		" AND sn.`classes%i` <= '%i'",
-		bot_inst->GetBotID(),
-		(DisciplineReuseStart - 1),
-		(DisciplineReuseStart - 1),
-		bot_inst->GetClass(),
-		bot_inst->GetLevel()
+		"SELECT *"
```

Comment:
I'll look in to seeing about how to do this. Haven't messed with the repos yet so might take me a little bit once I get some free time to dabble. If need be and it looks good currently I can always move it after too if it's taking too long

Reply:
> I'll look in to seeing about how to do this. Haven't messed with the repos yet so might take me a little bit once I get some free time to dabble. If need be and it looks good currently I can always move it after too if it's taking too long

If you need help with this, just let me know.

Some input from @Akkadius on if it thinks it's necessary to use repositories on this initial pull request would be helpful as well.

Reply:
I'd say repositories aren't a hard requirement to get this PR to the finish line - but I guarantee you it would make your life a lot easier if you used them here and elsewhere

Reply:
https://docs.eqemu.io/developer/repositories/

Reply:
I did the legwork here for you and pushed up what all of this looks like with repositories. It simplifies the code quite a bit and makes it easier to read / reason about.

It does require a primary key on the table and your migration was missing any. I added two keys back to the table, `bot_id` and `timer_id`

Reply:
I think timer_id key will cause conflicts with any spells that have timer 0 for their own recasts, right?

Reply:
> I think timer_id key will cause conflicts with any spells that have timer 0 for their own recasts, right?

That's a good point, didn't do testing with timer IDs of `0`.

Reply:
That's why I removed the keys initially, but if they're required repositories I think the only way to ensure no unnecessary blocks is using bot_id, spell_id and item_id all together. Can throw timer_id in there too if necessary